packages/xo-web/src/xo-app/new-vm/index.js

```diff
@@ -1329,6 +1331,34 @@ export default class NewVm extends BaseComponent {
 
   // DISKS -----------------------------------------------------------------------
 
+  _getDiskSrs = createSelector(
+    () => this.state.state.existingDisks,
+    () => this.state.state.VDIs,
+    (existingDisks, vdis) => {
+      const getSr = _ => _.$SR || _.SR
+      return uniqBy([...existingDisks, ...vdis], getSr).map(getSr)
+    }
+  )
+
+  _srsNotOnSameHost = createSelector(
+    this._getDiskSrs,
+    () => this.props.srs,
+    (diskSrs, srs) => {
+      let container
+      let sr
+      return diskSrs.some(srId => {
+        sr = srs[srId]
```

Comment:
`const sr`

Reply:
I avoided creating a variable in the loop.

Reply:
I don't see the gain of performance that we'll gain but i prefer declaring variables in the spot when it will be used. I will let @pdonias  give us  his opinion about it.

Reply:
Yes, using the same variable every time would be more performant but it wouldn't make a big difference anyway.